Issue109 write unit test using get random order

KangmingL opened the pull request:
Add test3 (use get_random_order to generate orders) to test_order_matching.py.

DanilYachmenev commented on defi/dao_cross/test/test_order_matching.py:
```diff
@@ -175,6 +175,54 @@ def test2(self) -> None:
+            expected_signature,
+            dedent=True,
+            fuzzy_match=True,
+        )
```
This is the only file that needs to be checked in
Remove other files from PR (not from master though)

DanilYachmenev commented on defi/dao_cross/test/tmp.pytest.log:
```diff
@@ -0,0 +1,4 @@
+11:11:05 - [36mINFO [0m hunit_test.py setUp:1042                               11:11:05 - [36mINFO [0m hunit_test.py setUp:1042                               
+################################################################################
+TestMatchOrders1.test3
+################################################################################
```
all the files with  `tmp.` at name are temporary, you never check them in
you can just remove all of them

DanilYachmenev commented on devops/compose/docker-compose.yml:
```diff
@@ -0,0 +1,98 @@
+version: '3'
+
+services:
+  base_app:
+    cap_add:
```
And this is the file that should stay in your local repo, do not check it in this PR

DanilYachmenev commented on defi/dao_cross/test/test_order_matching.py:
```diff
@@ -175,6 +175,54 @@ def test2(self) -> None:
+    
+    def test3(self) -> None:
+        orders = []
+
```
Do not leave empty space between the lines
Either remove or fill with a meaningful comment

DanilYachmenev commented on defi/dao_cross/test/test_order_matching.py:
```diff
@@ -175,6 +175,54 @@ def test2(self) -> None:
+        orders = []
+
+        for i in range(6):
+            order = ddacrord.get_random_order()
```
plant i as a `seed` in the function so its outcome is always the same.
We need it in unit tests because we can't actually generate random orders all the time so this way we retain the behaviour 

so will be like
```
order = ddacrord.get_random_order(seed=i)
```

DanilYachmenev commented on defi/dao_cross/test/test_order_matching.py:
```diff
@@ -175,6 +175,54 @@ def test2(self) -> None:
+        columns=token,amount,from,to
+        shape=(8, 4)
+        token  amount  from  to
+        0   ETH       6     0   3
+        1   BTC       6     3   0
+        2   ETH       3    -2   2
+        ...
+        5   BTC       1     0  -2
+        6   ETH       2    -3   0
+        7   BTC       2     0  -3
```
check couple of rows by hand to confirm the signature (will be updated after suggested changes)

DanilYachmenev commented on defi/dao_cross/test/test_order_matching.py:
```diff
@@ -175,6 +175,54 @@ def test2(self) -> None:
             fuzzy_match=True,
         )
+    
+    def test3(self) -> None:
```
Add a docstring
```
"""
All orders are randomly generated.
"""
```

DanilYachmenev commented on defi/dao_cross/test/test_order_matching.py:
```diff
@@ -175,6 +175,54 @@ def test2(self) -> None:
+            print_shape_info=True,
+            tag="df",
+        )
+        expected_signature = r"""
```
you can get `expected_signature` from error logs after you run the test
`>pytest  defi/dao_cross/test/test_order_matching.py -s --dbg`

DanilYachmenev reviewed (commented):
lg, see the comments for the further steps
overall, keep in mind that for now you have to check in only the files that you've changed

To run and check that the test is correct see the issue specs

DanilYachmenev commented:
> Add test3 (use get_random_order to generate orders) to test_order_matching.py.

And let's add a number of issue in the PR text body

#109 
This eases the navigation through the files

KangmingL commented:
@DanilYachmenev Thanks for your feedback. I made some changes to my branch. I removed all the tmp files and docker-compose, added seed to get_random_order(), and double-checked the signature by hand. I think it should be good to go.

DanilYachmenev commented on defi/dao_cross/test/outcomes/TestMatchOrders1.test3/tmp.final.actual.txt:
```diff
@@ -0,0 +1,12 @@
+# df=
```
so no need in this file

DanilYachmenev commented on defi/dao_cross/test/outcomes/TestMatchOrders1.test3/tmp.final.expected.txt:
```diff
@@ -0,0 +1,12 @@
+# df=
```
and no ned for this as well

DanilYachmenev commented on defi/dao_cross/test/test_order_matching.py:
```diff
@@ -175,6 +175,57 @@ def test2(self) -> None:
+    
+    def test3(self) -> None:
+        """
+        All orders are randomly generated
```
Always put a dot at the end of a dosctring `.`

DanilYachmenev reviewed (commented):
nice, couple nits and ready

and btw, could you run the test from your bash? did it pass?
